src/Classes/TradeQuery.lua

```diff
@@ -0,0 +1,897 @@
+	end)
+	main:OpenPopup(pane_width, pane_height, "Build Pricer", controls)
+
+	if #self.itemsTab.leagueDropList == 0 then
+		self:PullLeagueList(controls)
+	else
+		controls.league.selIndex = 1
+		self.pbLeague = self.itemsTab.leagueDropList[controls.league.selIndex].name
+		self.pbLeagueRealName = self.itemsTab.leagueDropList[controls.league.selIndex].realname
+		self:SetCurrencyConversionButton(controls)
+	end
+end
+
+-- Method to update the Currency Conversion button label
+function TradeQueryClass:SetCurrencyConversionButton(controls)
+	local currencyLabel = colorCodes.WARNING .. "Update Currency Conversion Rates"
+	self.pbFileTimestampDiff[controls.league.selIndex] = nil
+	local foo = io.open("../"..self.pbLeague.."_currency_values.json", "r")
+	if foo then
+		local lines = foo:read "*a"
+		foo:close()
+		self.pbCurrencyConversion[self.pbLeague] = self:ProcessJSON(lines)
+		self.lastCurrencyFileTime[controls.league.selIndex]  = self.pbCurrencyConversion[self.pbLeague]["updateTime"]
+		self.pbFileTimestampDiff[controls.league.selIndex] = get_time() - self.lastCurrencyFileTime[controls.league.selIndex] 
+		if self.pbFileTimestampDiff[controls.league.selIndex] < 3600 then
+			-- Less than 1 hour (60 * 60 = 3600)
+			currencyLabel = "^8Currency Rates are Very Recent"
+		elseif self.pbFileTimestampDiff[controls.league.selIndex] < (24 * 3600) then
+			-- Less than 1 day
+			currencyLabel = "^7Currency Rates are Recent"
+		end
+	else
+		currencyLabel = colorCodes.NEGATIVE .. "Get Currency Conversion Rates"
+	end
+	controls.updateCurrencyConversion.label = currencyLabel
+	controls.updateCurrencyConversion.enabled = function()
+		return self.pbFileTimestampDiff[controls.league.selIndex] == nil or self.pbFileTimestampDiff[controls.league.selIndex] >= 3600
+	end
+	controls.updateCurrencyConversion.tooltipFunc = function(tooltip)
+		tooltip:Clear()
+		if self.lastCurrencyFileTime[controls.league.selIndex] ~= nil then
+			self.pbFileTimestampDiff[controls.league.selIndex] = get_time() - self.lastCurrencyFileTime[controls.league.selIndex] 
+		end
+		if self.pbFileTimestampDiff[controls.league.selIndex] == nil or self.pbFileTimestampDiff[controls.league.selIndex] >= 3600 then
+			tooltip:AddLine(16, colorCodes.WARNING .. "Currency Conversion rates are pulled from PoE Ninja leveraging their API.")
+			tooltip:AddLine(16, colorCodes.WARNING .. "Updates are limited to once per hour and not necessary more than once per day.")
+			tooltip:AddLine(16, "")
+			tooltip:AddLine(16, colorCodes.NEGATIVE .. "NOTE: This will expose your IP address to poe.ninja.")
+			tooltip:AddLine(16, colorCodes.NEGATIVE .. "If you are concerned about this please do not click this button.")
+		elseif self.pbFileTimestampDiff[controls.league.selIndex] ~= nil and self.pbFileTimestampDiff[controls.league.selIndex] < 3600 then
+			tooltip:AddLine(16, "Conversion Rates are less than an hour old (" .. tostring(self.pbFileTimestampDiff[controls.league.selIndex]) .. " seconds old)")
```

Comment:
Ideally this would update to minutes at some point (doesn't need second-by-second granularity after the first minute), since 12345 seconds is hard to read.

Reply:
Sure, that could be a generic function when dealing with time to do proper conversion.